### ISSUE TITLE
Using MLFlow with BERTopic

### ISSUE THREAD

**Issue by doubianimehdi** [MaartenGr/BERTopic#234]
Hi,

Thanks again for your help and your responsiveness.
I was wondering if you could give us some code snippet to use MLFlow with BERTopic to be able to compare the quality of the modeling with different embeddings/hyperparameters.

Thanks !

**Comment by MaartenGr** [comment]
For MLFlow specifically, I do not have any code laying around. However, you can follow along with [Gensim](https://radimrehurek.com/gensim/models/coherencemodel.html) or [OCTIS](https://github.com/MIND-Lab/OCTIS) for nice evaluation schemes. 

When it comes to evaluating the quality of a topic model I am wondering if you are referring to specific evaluation criteria. Did you have something specific in mind? 

The thing is that evaluating and tuning a somewhat subjective technique such as topic modeling comes with its risks since the evaluation measures can have significant flaws. For example, if you are optimizing for any evaluation metric, there is a good chance that a higher value doesn't necessarily mean a better topic model. Because of the subjective nature of topic modeling, defining "quality" is rather difficult since individuals may disagree about the quality of topic representations. 

In other words, when optimizing the hyperparameters of BERTopic, keep in mind what you are optimizing it for. 

Hopefully, this helps a bit!

**Comment by nilsblessing** [comment]
@doubianimehdi in case you know your metrics of interest you can refer to the [mlflow documentation](https://www.mlflow.org/docs/latest/tracking.html#logging-data-to-runs) to log and compare custom parameters and metrics within an experiment.

kind regards
Nils

**Comment by MaartenGr** [comment]
I am closing this now due to inactivity. However, if you are still experiencing this issue or another, please let know!
